modules/oe_editorial_unpublish/src/Form/NodeUnpublishForm.php

```diff
@@ -0,0 +1,228 @@
+    $this->eventDispatcher = $event_dispatcher;
+    $this->moderationInfo = $moderation_info;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('event_dispatcher'),
+      $container->get('content_moderation.moderation_information')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId(): string {
+    return 'node_unpublish_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, NodeInterface $node = NULL): array {
+    $this->node = $node;
+    $form = parent::buildForm($form, $form_state);
+    $workflow = $this->moderationInfo->getWorkflowForEntity($node);
+    $unpublished_states = $this->getUnpublishableStates($workflow->getTypePlugin(), $node);
+    $unpublished_states = array_map(function ($state) {
+      return $state->label();
+    }, $unpublished_states);
+
+    $form['unpublish_state'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Select the state to unpublish this node'),
+      '#options' => $unpublished_states,
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+    $this->node->moderation_state->value = $form_state->getValue('unpublish_state');
+    $this->node->save();
+    $this->messenger()->addStatus($this->t('The node %label has been unpublished.', [
+      '%label' => $this->node->label(),
+    ]));
+    $form_state->setRedirectUrl($this->node->toUrl());
+
+  }
+
+  /**
+   * Checks access for a specific request.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The account accessing the route.
+   * @param \Drupal\node\NodeInterface $node
+   *   The node being accessed.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function access(AccountInterface $account, NodeInterface $node): AccessResultInterface {
+    if (!$this->moderationInfo->isModeratedEntity($node)) {
+      // If the content is not using content moderation, we deny access.
+      return AccessResult::forbidden($this->t('Content does not have content moderation enabled.'))->addCacheableDependency($node);
+    }
+
+    $storage = $this->entityTypeManager->getStorage($node->getEntityTypeId());
+    $latest_revision_id = $storage->getLatestTranslationAffectedRevisionId($node->id(), $node->language()->getId());
+    if ($latest_revision_id === NULL || !$this->moderationInfo->isDefaultRevisionPublished($node)) {
+      // The content does not have a published version.
+      return AccessResult::forbidden($this->t('The content does not have a published version.'))->addCacheableDependency($node);
+    }
+
+    // Check if the user has a permission to transition to an unpublished state.
```

Comment:
Now the comment is obsolete, since you are doing the actual check in the getUnpublishableStates().

Reply:
Moved the comment and reworded the accessdenied message.